### PR TITLE
Issue #9567: update example of AST for TokenTypes.SEMI

### PR DESCRIPTION
**Pull request by SGanguly1999**
Closes: #9567

image

`$ cat Demo.java`

```
class Demo
{
    public static void main(String args[])
    {
        for(;;);
    }
}
```

`$ java -jar checkstyle-8.39-all.jar -t Demo.java`

```
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|--LITERAL_CLASS -> class [1:0]
|--IDENT -> Demo [1:6]
`--OBJBLOCK -> OBJBLOCK [2:0]
    |--LCURLY -> { [2:0]
    |--METHOD_DEF -> METHOD_DEF [3:4]
    |   |--MODIFIERS -> MODIFIERS [3:4]
    |   |   |--LITERAL_PUBLIC -> public [3:4]
    |   |   `--LITERAL_STATIC -> static [3:11]
    |   |--TYPE -> TYPE [3:18]
    |   |   `--LITERAL_VOID -> void [3:18]
    |   |--IDENT -> main [3:23]
    |   |--LPAREN -> ( [3:27]
    |   |--PARAMETERS -> PARAMETERS [3:39]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [3:39]
    |   |       |--MODIFIERS -> MODIFIERS [3:39]
    |   |       |--TYPE -> TYPE [3:39]
    |   |       |   `--ARRAY_DECLARATOR -> [ [3:39]
    |   |       |       |--IDENT -> String [3:28]
    |   |       |       `--RBRACK -> ] [3:40]
    |   |       `--IDENT -> args [3:35]
    |   |--RPAREN -> ) [3:41]
    |   `--SLIST -> { [4:4]
    |       |--LITERAL_FOR -> for [5:8]
    |       |   |--LPAREN -> ( [5:11]
    |       |   |--FOR_INIT -> FOR_INIT [5:12]
    |       |   |--SEMI -> ; [5:12]
    |       |   |--FOR_CONDITION -> FOR_CONDITION [5:13]
    |       |   |--SEMI -> ; [5:13]
    |       |   |--FOR_ITERATOR -> FOR_ITERATOR [5:14]
    |       |   |--RPAREN -> ) [5:14]
    |       |   `--EMPTY_STAT -> ; [5:15]
    |       `--RCURLY -> } [6:4]
    `--RCURLY -> } [7:0]

```
`$ java -jar checkstyle-8.38-all.jar -t Demo.java | grep "5:"`

```
 LITERAL_FOR -> for [5:8]
 |--LPAREN -> ( [5:11]
 |--FOR_INIT -> FOR_INIT [5:12]
 |--SEMI -> ; [5:12]
 |--FOR_CONDITION -> FOR_CONDITION [5:13]
 |--SEMI -> ; [5:13]
 |--FOR_ITERATOR -> FOR_ITERATOR [5:14]
 |--RPAREN -> ) [5:14]
 `--EMPTY_STAT -> ; [5:15]
```

Expected update for Javadoc:

```
LITERAL_FOR -> for
  |--LPAREN -> (
  |--FOR_INIT -> FOR_INIT
  |--SEMI -> ;
  |--FOR_CONDITION -> FOR_CONDITION
  |--SEMI -> ;
  |--FOR_ITERATOR -> FOR_ITERATOR
  |--RPAREN -> )
  `--EMPTY_STAT -> ;

```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1130,6 +1130,21 @@
      * The statement terminator ({@code ;}).  Depending on the
      * context, this make occur as a sibling, a child, or not at all.
      *
+     * <p>For example:</p>
+     * <pre>boolean var = true;</pre>
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1134]: please, make tags and content on separate lines, like in [here](https://github.com/checkstyle/checkstyle/pull/9378/files#diff-ba93830ba2cf23d542b30560fb43bdc8d8d1be05025e84ee6a96cca6ef8e3342R1430)

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1134]: This example is too big. I suggest to use something like
```java
       for (;;)
        ;
```
instead.

**Review by pbludov — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (SGanguly1999) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @timurt done

Comment (SGanguly1999) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @pbludov done

**Review by timurt — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Pull request merged**